src/misc/lv_obj_property.c

```diff
@@ -0,0 +1,179 @@
+/**
+ * @file lv_obj_id.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "../core/lv_obj.h"
+#include "../stdlib/lv_string.h"
+#include "lv_obj_property.h"
+
+#if LV_USE_OBJ_PROPERTY
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+typedef void (*lv_property_set_int_t)(const struct _lv_obj_t *, int32_t);
+typedef void (*lv_property_set_pointer_t)(const struct _lv_obj_t *, const void *);
+typedef void (*lv_property_set_color_t)(const struct _lv_obj_t *, lv_color_t);
+typedef lv_result_t (*lv_property_setter_t)(const struct _lv_obj_t *, lv_prop_id_t, const lv_property_t *);
+
+typedef int32_t (*lv_property_get_int_t)(const struct _lv_obj_t *);
+typedef void * (*lv_property_get_pointer_t)(const struct _lv_obj_t *);
+typedef lv_color_t (*lv_property_get_color_t)(const struct _lv_obj_t *);
+typedef lv_result_t (*lv_property_getter_t)(const struct _lv_obj_t *, lv_prop_id_t, lv_property_t *);
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+static lv_result_t obj_property(lv_obj_t * obj, lv_prop_id_t id, lv_property_t * value, bool set);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+lv_result_t lv_obj_property_set(lv_obj_t * obj, const lv_property_t * value)
```

Comment:
What about this?
```c
lv_result_t lv_obj_set_property(lv_obj_t * obj, uint32_t id, const lv_value_group_t value)
```

`lv_value_group_t` could be used here, in styles and in [lv_observer](https://github.com/lvgl/lvgl/blob/master/src/others/observer/lv_observer.c) as well.

Reply:
Let's add `lv_value_group_t` in another PR.